python3Package.llm: Build fails on darwin

### Nixpkgs version

- Unstable (26.05)

### Describe the bug

`python3Packages.llm` fails to build on darwin. This happens on nixpkgs-unstable and master.

### Steps to reproduce

```
❯ git switch nixpkgs-untable
❯ git pull
❯ nix build --print-build-logs .#python3Packages.llm
```

### Expected behaviour

The build should not fail

### Screenshots

_No response_

### Relevant log output

```console
python3.13-llm> =================================== FAILURES ===================================
python3.13-llm> ________________ test_logs_fragments[fragment_refs0-expected0] _________________
python3.13-llm> fragments_fixture = {'collected': {'both_fragments_id': {'both_fragments': '01kdzm6001tt83cyje1exn7mq3'}, 'multi_prompt_fragment_id': {'mu...r1i8yzaxkc6m88ip7qnsda/pytest-of-_nixbld1/pytest-0/test_logs_fragments_fragment_r0/llm.datasette.io/logs_fragments.db'}
python3.13-llm> fragment_refs = ['hash1']
python3.13-llm> expected = [{'name': 'single_prompt_fragment', 'prompt_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d...60a19107f13dd6cdcdc78c577b71c58f645d77ec95015e3', '7328249eecee2ec90891237050a7283050bd4560bc5d0bf3643663d340d44cfa']}]
python3.13-llm>     @pytest.mark.parametrize(
python3.13-llm>         "fragment_refs,expected",
python3.13-llm>         (
python3.13-llm>             (
python3.13-llm>                 ["hash1"],
python3.13-llm>                 [
python3.13-llm>                     {
python3.13-llm>                         "name": "single_prompt_fragment",
python3.13-llm>                         "prompt_fragments": ["hash1"],
python3.13-llm>                         "system_fragments": [],
python3.13-llm>                     },
python3.13-llm>                     {
python3.13-llm>                         "name": "multi_prompt_fragment",
python3.13-llm>                         "prompt_fragments": ["hash1", "hash2"],
python3.13-llm>                         "system_fragments": [],
python3.13-llm>                     },
python3.13-llm>                     {
python3.13-llm>                         "name": "multi_system_fragment",
python3.13-llm>                         "prompt_fragments": [],
python3.13-llm>                         "system_fragments": ["hash1", "hash2"],
python3.13-llm>                     },
python3.13-llm>                     {
python3.13-llm>                         "name": "both_fragments",
python3.13-llm>                         "prompt_fragments": ["hash1", "hash2"],
python3.13-llm>                         "system_fragments": ["hash3", "hash4"],
python3.13-llm>                     },
python3.13-llm>                 ],
python3.13-llm>             ),
python3.13-llm>             (
python3.13-llm>                 ["alias_3"],
python3.13-llm>                 [
python3.13-llm>                     {
python3.13-llm>                         "name": "both_fragments",
python3.13-llm>                         "prompt_fragments": ["hash1", "hash2"],
python3.13-llm>                         "system_fragments": ["hash3", "hash4"],
python3.13-llm>                     },
python3.13-llm>                     {
python3.13-llm>                         "name": "single_system_fragment_with_alias",
python3.13-llm>                         "prompt_fragments": [],
python3.13-llm>                         "system_fragments": ["hash4"],
python3.13-llm>                     },
python3.13-llm>                 ],
python3.13-llm>             ),
python3.13-llm>             # Testing for AND condition
python3.13-llm>             (
python3.13-llm>                 ["hash1", "hash4"],
python3.13-llm>                 [
python3.13-llm>                     {
python3.13-llm>                         "name": "both_fragments",
python3.13-llm>                         "prompt_fragments": ["hash1", "hash2"],
python3.13-llm>                         "system_fragments": ["hash3", "hash4"],
python3.13-llm>                     },
python3.13-llm>                 ],
python3.13-llm>             ),
python3.13-llm>         ),
python3.13-llm>     )
python3.13-llm>     def test_logs_fragments(fragments_fixture, fragment_refs, expected):
python3.13-llm>         fragments_log_path = fragments_fixture["path"]
python3.13-llm>         fragment_hashes_by_slug = fragments_fixture["fragment_hashes_by_slug"]
python3.13-llm>         runner = CliRunner()
python3.13-llm>         args = ["logs", "-d", fragments_log_path, "-n", "0"]
python3.13-llm>         for ref in fragment_refs:
python3.13-llm>             args.extend(["-f", ref])
python3.13-llm>         result = runner.invoke(cli, args + ["--json"], catch_exceptions=False)
python3.13-llm>         assert result.exit_code == 0
python3.13-llm>         output = result.output
python3.13-llm>         responses = json.loads(output)
python3.13-llm>         # Re-shape that to same shape as expected
python3.13-llm>         reshaped = [
python3.13-llm>             {
python3.13-llm>                 "name": response["prompt"].replace("prompt: ", ""),
python3.13-llm>                 "prompt_fragments": [
python3.13-llm>                     fragment["hash"] for fragment in response["prompt_fragments"]
python3.13-llm>                 ],
python3.13-llm>                 "system_fragments": [
python3.13-llm>                     fragment["hash"] for fragment in response["system_fragments"]
python3.13-llm>                 ],
python3.13-llm>             }
python3.13-llm>             for response in responses
python3.13-llm>         ]
python3.13-llm>         # Replace aliases with hash IDs in expected
python3.13-llm>         for item in expected:
python3.13-llm>             item["prompt_fragments"] = [
python3.13-llm>                 fragment_hashes_by_slug.get(ref, ref) for ref in item["prompt_fragments"]
python3.13-llm>             ]
python3.13-llm>             item["system_fragments"] = [
python3.13-llm>                 fragment_hashes_by_slug.get(ref, ref) for ref in item["system_fragments"]
python3.13-llm>             ]
python3.13-llm> >       assert reshaped == expected
python3.13-llm> E       AssertionError: assert [{'name': 'multi_system_fragment', 'prompt_fragments': [], 'system_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb', '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6']}] == [{'name': 'single_prompt_fragment', 'prompt_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb'], 'system_fragments': []}, {'name': 'multi_prompt_fragment', 'prompt_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb', '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6'], 'system_fragments': []}, {'name': 'multi_system_fragment', 'prompt_fragments': [], 'system_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb', '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6']}, {'name': 'both_fragments', 'prompt_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb', '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6'], 'system_fragments': ['de2dc9fbffbfd7e3760a19107f13dd6cdcdc78c577b71c58f645d77ec95015e3', '7328249eecee2ec90891237050a7283050bd4560bc5d0bf3643663d340d44cfa']}]
python3.13-llm> E
python3.13-llm> E         At index 0 diff: {'name': 'multi_system_fragment', 'prompt_fragments': [], 'system_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb', '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6']} != {'name': 'single_prompt_fragment', 'prompt_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb'], 'system_fragments': []}
python3.13-llm> E         Right contains 3 more items, first extra item: {'name': 'multi_prompt_fragment', 'prompt_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb', '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6'], 'system_fragments': []}
python3.13-llm> E
python3.13-llm> E         Full diff:
python3.13-llm> E           [
python3.13-llm> E         -     {
python3.13-llm> E         -         'name': 'single_prompt_fragment',
python3.13-llm> E         -         'prompt_fragments': [
python3.13-llm> E         -             'b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb',
python3.13-llm> E         -         ],
python3.13-llm> E         -         'system_fragments': [],
python3.13-llm> E         -     },
python3.13-llm> E         -     {
python3.13-llm> E         -         'name': 'multi_prompt_fragment',
python3.13-llm> E         -         'prompt_fragments': [
python3.13-llm> E         -             'b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb',
python3.13-llm> E         -             '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6',
python3.13-llm> E         -         ],
python3.13-llm> E         -         'system_fragments': [],
python3.13-llm> E         -     },
python3.13-llm> E               {
python3.13-llm> E                   'name': 'multi_system_fragment',
python3.13-llm> E                   'prompt_fragments': [],
python3.13-llm> E                   'system_fragments': [
python3.13-llm> E                       'b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb',
python3.13-llm> E                       '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6',
python3.13-llm> E                   ],
python3.13-llm> E               },
python3.13-llm> E         -     {
python3.13-llm> E         -         'name': 'both_fragments',
python3.13-llm> E         -         'prompt_fragments': [
python3.13-llm> E         -             'b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb',
python3.13-llm> E         -             '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6',
python3.13-llm> E         -         ],
python3.13-llm> E         -         'system_fragments': [
python3.13-llm> E         -             'de2dc9fbffbfd7e3760a19107f13dd6cdcdc78c577b71c58f645d77ec95015e3',
python3.13-llm> E         -             '7328249eecee2ec90891237050a7283050bd4560bc5d0bf3643663d340d44cfa',
python3.13-llm> E         -         ],
python3.13-llm> E         -     },
python3.13-llm> E           ]
python3.13-llm> tests/test_llm_logs.py:661: AssertionError
python3.13-llm> ________________ test_logs_fragments[fragment_refs2-expected2] _________________
python3.13-llm> fragments_fixture = {'collected': {'both_fragments_id': {'both_fragments': '01kdzm624gpbazw1zy3y6w3q0z'}, 'multi_prompt_fragment_id': {'mu...r1i8yzaxkc6m88ip7qnsda/pytest-of-_nixbld1/pytest-0/test_logs_fragments_fragment_r2/llm.datasette.io/logs_fragments.db'}
python3.13-llm> fragment_refs = ['hash1', 'hash4']
python3.13-llm> expected = [{'name': 'both_fragments', 'prompt_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb', '...60a19107f13dd6cdcdc78c577b71c58f645d77ec95015e3', '7328249eecee2ec90891237050a7283050bd4560bc5d0bf3643663d340d44cfa']}]
python3.13-llm>     @pytest.mark.parametrize(
python3.13-llm>         "fragment_refs,expected",
python3.13-llm>         (
python3.13-llm>             (
python3.13-llm>                 ["hash1"],
python3.13-llm>                 [
python3.13-llm>                     {
python3.13-llm>                         "name": "single_prompt_fragment",
python3.13-llm>                         "prompt_fragments": ["hash1"],
python3.13-llm>                         "system_fragments": [],
python3.13-llm>                     },
python3.13-llm>                     {
python3.13-llm>                         "name": "multi_prompt_fragment",
python3.13-llm>                         "prompt_fragments": ["hash1", "hash2"],
python3.13-llm>                         "system_fragments": [],
python3.13-llm>                     },
python3.13-llm>                     {
python3.13-llm>                         "name": "multi_system_fragment",
python3.13-llm>                         "prompt_fragments": [],
python3.13-llm>                         "system_fragments": ["hash1", "hash2"],
python3.13-llm>                     },
python3.13-llm>                     {
python3.13-llm>                         "name": "both_fragments",
python3.13-llm>                         "prompt_fragments": ["hash1", "hash2"],
python3.13-llm>                         "system_fragments": ["hash3", "hash4"],
python3.13-llm>                     },
python3.13-llm>                 ],
python3.13-llm>             ),
python3.13-llm>             (
python3.13-llm>                 ["alias_3"],
python3.13-llm>                 [
python3.13-llm>                     {
python3.13-llm>                         "name": "both_fragments",
python3.13-llm>                         "prompt_fragments": ["hash1", "hash2"],
python3.13-llm>                         "system_fragments": ["hash3", "hash4"],
python3.13-llm>                     },
python3.13-llm>                     {
python3.13-llm>                         "name": "single_system_fragment_with_alias",
python3.13-llm>                         "prompt_fragments": [],
python3.13-llm>                         "system_fragments": ["hash4"],
python3.13-llm>                     },
python3.13-llm>                 ],
python3.13-llm>             ),
python3.13-llm>             # Testing for AND condition
python3.13-llm>             (
python3.13-llm>                 ["hash1", "hash4"],
python3.13-llm>                 [
python3.13-llm>                     {
python3.13-llm>                         "name": "both_fragments",
python3.13-llm>                         "prompt_fragments": ["hash1", "hash2"],
python3.13-llm>                         "system_fragments": ["hash3", "hash4"],
python3.13-llm>                     },
python3.13-llm>                 ],
python3.13-llm>             ),
python3.13-llm>         ),
python3.13-llm>     )
python3.13-llm>     def test_logs_fragments(fragments_fixture, fragment_refs, expected):
python3.13-llm>         fragments_log_path = fragments_fixture["path"]
python3.13-llm>         fragment_hashes_by_slug = fragments_fixture["fragment_hashes_by_slug"]
python3.13-llm>         runner = CliRunner()
python3.13-llm>         args = ["logs", "-d", fragments_log_path, "-n", "0"]
python3.13-llm>         for ref in fragment_refs:
python3.13-llm>             args.extend(["-f", ref])
python3.13-llm>         result = runner.invoke(cli, args + ["--json"], catch_exceptions=False)
python3.13-llm>         assert result.exit_code == 0
python3.13-llm>         output = result.output
python3.13-llm>         responses = json.loads(output)
python3.13-llm>         # Re-shape that to same shape as expected
python3.13-llm>         reshaped = [
python3.13-llm>             {
python3.13-llm>                 "name": response["prompt"].replace("prompt: ", ""),
python3.13-llm>                 "prompt_fragments": [
python3.13-llm>                     fragment["hash"] for fragment in response["prompt_fragments"]
python3.13-llm>                 ],
python3.13-llm>                 "system_fragments": [
python3.13-llm>                     fragment["hash"] for fragment in response["system_fragments"]
python3.13-llm>                 ],
python3.13-llm>             }
python3.13-llm>             for response in responses
python3.13-llm>         ]
python3.13-llm>         # Replace aliases with hash IDs in expected
python3.13-llm>         for item in expected:
python3.13-llm>             item["prompt_fragments"] = [
python3.13-llm>                 fragment_hashes_by_slug.get(ref, ref) for ref in item["prompt_fragments"]
python3.13-llm>             ]
python3.13-llm>             item["system_fragments"] = [
python3.13-llm>                 fragment_hashes_by_slug.get(ref, ref) for ref in item["system_fragments"]
python3.13-llm>             ]
python3.13-llm> >       assert reshaped == expected
python3.13-llm> E       AssertionError: assert [] == [{'name': 'both_fragments', 'prompt_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb', '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6'], 'system_fragments': ['de2dc9fbffbfd7e3760a19107f13dd6cdcdc78c577b71c58f645d77ec95015e3', '7328249eecee2ec90891237050a7283050bd4560bc5d0bf3643663d340d44cfa']}]
python3.13-llm> E
python3.13-llm> E         Right contains one more item: {'name': 'both_fragments', 'prompt_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb', '2...760a19107f13dd6cdcdc78c577b71c58f645d77ec95015e3', '7328249eecee2ec90891237050a7283050bd4560bc5d0bf3643663d340d44cfa']}
python3.13-llm> E
python3.13-llm> E         Full diff:
python3.13-llm> E         + []
python3.13-llm> E         - [
python3.13-llm> E         -     {
python3.13-llm> E         -         'name': 'both_fragments',
python3.13-llm> E         -         'prompt_fragments': [
python3.13-llm> E         -             'b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb',
python3.13-llm> E         -             '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6',
python3.13-llm> E         -         ],
python3.13-llm> E         -         'system_fragments': [
python3.13-llm> E         -             'de2dc9fbffbfd7e3760a19107f13dd6cdcdc78c577b71c58f645d77ec95015e3',
python3.13-llm> E         -             '7328249eecee2ec90891237050a7283050bd4560bc5d0bf3643663d340d44cfa',
python3.13-llm> E         -         ],
python3.13-llm> E         -     },
python3.13-llm> E         - ]
python3.13-llm> tests/test_llm_logs.py:661: AssertionError
python3.13-llm> ________________________ test_expand_fragment_json[-e] _________________________
python3.13-llm> fragments_fixture = {'collected': {'both_fragments_id': {'both_fragments': '01kdzm6423e52ccp63sredszpr'}, 'multi_prompt_fragment_id': {'mu...jr1i8yzaxkc6m88ip7qnsda/pytest-of-_nixbld1/pytest-0/test_expand_fragment_json__e_0/llm.datasette.io/logs_fragments.db'}
python3.13-llm> arg = '-e'
python3.13-llm>     @pytest.mark.parametrize("arg", ("-e", "--expand"))
python3.13-llm>     def test_expand_fragment_json(fragments_fixture, arg):
python3.13-llm>         fragments_log_path = fragments_fixture["path"]
python3.13-llm>         runner = CliRunner()
python3.13-llm>         args = ["logs", "-d", fragments_log_path, "-f", "long_5", "--json"]
python3.13-llm>         # Without -e the JSON is truncated
python3.13-llm>         result = runner.invoke(cli, args, catch_exceptions=False)
python3.13-llm>         assert result.exit_code == 0
python3.13-llm>         data = json.loads(result.output)
python3.13-llm> >       fragment = data[0]["prompt_fragments"][0]["content"]
python3.13-llm>                    ^^^^^^^
python3.13-llm> E       IndexError: list index out of range
python3.13-llm> tests/test_llm_logs.py:869: IndexError
python3.13-llm> _____________________ test_expand_fragment_json[--expand] ______________________
python3.13-llm> fragments_fixture = {'collected': {'both_fragments_id': {'both_fragments': '01kdzm652awxgwrk72zxab42h5'}, 'multi_prompt_fragment_id': {'mu...r1i8yzaxkc6m88ip7qnsda/pytest-of-_nixbld1/pytest-0/test_expand_fragment_json___ex0/llm.datasette.io/logs_fragments.db'}
python3.13-llm> arg = '--expand'
python3.13-llm>     @pytest.mark.parametrize("arg", ("-e", "--expand"))
python3.13-llm>     def test_expand_fragment_json(fragments_fixture, arg):
python3.13-llm>         fragments_log_path = fragments_fixture["path"]
python3.13-llm>         runner = CliRunner()
python3.13-llm>         args = ["logs", "-d", fragments_log_path, "-f", "long_5", "--json"]
python3.13-llm>         # Without -e the JSON is truncated
python3.13-llm>         result = runner.invoke(cli, args, catch_exceptions=False)
python3.13-llm>         assert result.exit_code == 0
python3.13-llm>         data = json.loads(result.output)
python3.13-llm> >       fragment = data[0]["prompt_fragments"][0]["content"]
python3.13-llm>                    ^^^^^^^
python3.13-llm> E       IndexError: list index out of range
python3.13-llm> tests/test_llm_logs.py:869: IndexError
python3.13-llm> ________________________ test_expand_fragment_markdown _________________________
python3.13-llm> fragments_fixture = {'collected': {'both_fragments_id': {'both_fragments': '01kdzm6631kjnpdj78h73xfx9h'}, 'multi_prompt_fragment_id': {'mu...jr1i8yzaxkc6m88ip7qnsda/pytest-of-_nixbld1/pytest-0/test_expand_fragment_markdown0/llm.datasette.io/logs_fragments.db'}
python3.13-llm>     def test_expand_fragment_markdown(fragments_fixture):
python3.13-llm>         fragments_log_path = fragments_fixture["path"]
python3.13-llm>         fragment_hashes_by_slug = fragments_fixture["fragment_hashes_by_slug"]
python3.13-llm>         runner = CliRunner()
python3.13-llm>         args = ["logs", "-d", fragments_log_path, "-f", "long_5", "--expand"]
python3.13-llm>         result = runner.invoke(cli, args, catch_exceptions=False)
python3.13-llm>         assert result.exit_code == 0
python3.13-llm>         output = result.output
python3.13-llm>         interesting_bit = (
python3.13-llm> >           output.split("prompt: single_long_prompt_fragment_with_alias")[1]
python3.13-llm>             .split("## System")[0]
python3.13-llm>             .strip()
python3.13-llm>         )
python3.13-llm> E       IndexError: list index out of range
python3.13-llm> tests/test_llm_logs.py:890: IndexError
python3.13-llm> =========================== short test summary info ============================
python3.13-llm> FAILED tests/test_llm_logs.py::test_logs_fragments[fragment_refs0-expected0] - AssertionError: assert [{'name': 'multi_system_fragment', 'prompt_fragments': [], 'system_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb', '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6']}] == [{'name': 'single_prompt_fragment', 'prompt_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb'], 'system_fragments': []}, {'name': 'multi_prompt_fragment', 'prompt_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb', '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6'], 'system_fragments': []}, {'name': 'multi_system_fragment', 'prompt_fragments': [], 'system_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb', '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6']}, {'name': 'both_fragments', 'prompt_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb', '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6'], 'system_fragments': ['de2dc9fbffbfd7e3760a19107f13dd6cdcdc78c577b71c58f645d77ec95015e3', '7328249eecee2ec90891237050a7283050bd4560bc5d0bf3643663d340d44cfa']}]
python3.13-llm>   At index 0 diff: {'name': 'multi_system_fragment', 'prompt_fragments': [], 'system_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb', '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6']} != {'name': 'single_prompt_fragment', 'prompt_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb'], 'system_fragments': []}
python3.13-llm>   Right contains 3 more items, first extra item: {'name': 'multi_prompt_fragment', 'prompt_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb', '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6'], 'system_fragments': []}
python3.13-llm>   Full diff:
python3.13-llm>     [
python3.13-llm>   -     {
python3.13-llm>   -         'name': 'single_prompt_fragment',
python3.13-llm>   -         'prompt_fragments': [
python3.13-llm>   -             'b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb',
python3.13-llm>   -         ],
python3.13-llm>   -         'system_fragments': [],
python3.13-llm>   -     },
python3.13-llm>   -     {
python3.13-llm>   -         'name': 'multi_prompt_fragment',
python3.13-llm>   -         'prompt_fragments': [
python3.13-llm>   -             'b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb',
python3.13-llm>   -             '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6',
python3.13-llm>   -         ],
python3.13-llm>   -         'system_fragments': [],
python3.13-llm>   -     },
python3.13-llm>         {
python3.13-llm>             'name': 'multi_system_fragment',
python3.13-llm>             'prompt_fragments': [],
python3.13-llm>             'system_fragments': [
python3.13-llm>                 'b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb',
python3.13-llm>                 '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6',
python3.13-llm>             ],
python3.13-llm>         },
python3.13-llm>   -     {
python3.13-llm>   -         'name': 'both_fragments',
python3.13-llm>   -         'prompt_fragments': [
python3.13-llm>   -             'b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb',
python3.13-llm>   -             '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6',
python3.13-llm>   -         ],
python3.13-llm>   -         'system_fragments': [
python3.13-llm>   -             'de2dc9fbffbfd7e3760a19107f13dd6cdcdc78c577b71c58f645d77ec95015e3',
python3.13-llm>   -             '7328249eecee2ec90891237050a7283050bd4560bc5d0bf3643663d340d44cfa',
python3.13-llm>   -         ],
python3.13-llm>   -     },
python3.13-llm>     ]
python3.13-llm> FAILED tests/test_llm_logs.py::test_logs_fragments[fragment_refs2-expected2] - AssertionError: assert [] == [{'name': 'both_fragments', 'prompt_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb', '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6'], 'system_fragments': ['de2dc9fbffbfd7e3760a19107f13dd6cdcdc78c577b71c58f645d77ec95015e3', '7328249eecee2ec90891237050a7283050bd4560bc5d0bf3643663d340d44cfa']}]
python3.13-llm>   Right contains one more item: {'name': 'both_fragments', 'prompt_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb', '2...760a19107f13dd6cdcdc78c577b71c58f645d77ec95015e3', '7328249eecee2ec90891237050a7283050bd4560bc5d0bf3643663d340d44cfa']}
python3.13-llm>   Full diff:
python3.13-llm>   + []
python3.13-llm>   - [
python3.13-llm>   -     {
python3.13-llm>   -         'name': 'both_fragments',
python3.13-llm>   -         'prompt_fragments': [
python3.13-llm>   -             'b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb',
python3.13-llm>   -             '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6',
python3.13-llm>   -         ],
python3.13-llm>   -         'system_fragments': [
python3.13-llm>   -             'de2dc9fbffbfd7e3760a19107f13dd6cdcdc78c577b71c58f645d77ec95015e3',
python3.13-llm>   -             '7328249eecee2ec90891237050a7283050bd4560bc5d0bf3643663d340d44cfa',
python3.13-llm>   -         ],
python3.13-llm>   -     },
python3.13-llm>   - ]
python3.13-llm> FAILED tests/test_llm_logs.py::test_expand_fragment_json[-e] - IndexError: list index out of range
python3.13-llm> FAILED tests/test_llm_logs.py::test_expand_fragment_json[--expand] - IndexError: list index out of range
python3.13-llm> FAILED tests/test_llm_logs.py::test_expand_fragment_markdown - IndexError: list index out of range
python3.13-llm> ================ 5 failed, 453 passed, 11 deselected in 20.11s =================
error: builder for '/nix/store/qs5j0rnmsy6i19nx3vgqfdhnmzg5sy9b-python3.13-llm-0.28.drv' failed with exit code 1;
       last 25 log lines:
       >   -     },
       >     ]
       > FAILED tests/test_llm_logs.py::test_logs_fragments[fragment_refs2-expected2] - AssertionError: assert [] == [{'name': 'both_fragments', 'prompt_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb', '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6'], 'system_fragments': ['de2dc9fbffbfd7e3760a19107f13dd6cdcdc78c577b71c58f645d77ec95015e3', '7328249eecee2ec90891237050a7283050bd4560bc5d0bf3643663d340d44cfa']}]
       >
       >   Right contains one more item: {'name': 'both_fragments', 'prompt_fragments': ['b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb', '2...760a19107f13dd6cdcdc78c577b71c58f645d77ec95015e3', '7328249eecee2ec90891237050a7283050bd4560bc5d0bf3643663d340d44cfa']}
       >
       >   Full diff:
       >   + []
       >   - [
       >   -     {
       >   -         'name': 'both_fragments',
       >   -         'prompt_fragments': [
       >   -             'b4887c00c11d318678012a003fc53203dae1f7a7337e58c982485dbe055d50fb',
       >   -             '292244e938bf29eca2b7c584a4dab12f479293b2d8ff300343e1b61d30b5efc6',
       >   -         ],
       >   -         'system_fragments': [
       >   -             'de2dc9fbffbfd7e3760a19107f13dd6cdcdc78c577b71c58f645d77ec95015e3',
       >   -             '7328249eecee2ec90891237050a7283050bd4560bc5d0bf3643663d340d44cfa',
       >   -         ],
       >   -     },
       >   - ]
       > FAILED tests/test_llm_logs.py::test_expand_fragment_json[-e] - IndexError: list index out of range
       > FAILED tests/test_llm_logs.py::test_expand_fragment_json[--expand] - IndexError: list index out of range
       > FAILED tests/test_llm_logs.py::test_expand_fragment_markdown - IndexError: list index out of range
       > ================ 5 failed, 453 passed, 11 deselected in 20.11s =================
       For full logs, run:
            nix log /nix/store/qs5j0rnmsy6i19nx3vgqfdhnmzg5sy9b-python3.13-llm-0.28.drv
⏎
```

### Additional context

_No response_

### System metadata

❯ nix-shell -p nix-info --run "nix-info -m"
fetching path input 'path:/nix/store/89bzgr2smf8jdqlbay9apn6h81qrp238-source'
 - system: `"aarch64-darwin"`
 - host os: `Darwin 25.2.0, macOS 26.2`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Lix, like Nix) 2.94.0
System type: aarch64-darwin
Additional system types: aarch64-darwin, x86_64-darwin
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /Users/dwt/.config/nix/nix.conf:/Users/dwt/.nix-profile/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/bq9kpfvkpb18f3x2f9r6scxx94rbcymb-lix-2.94.0/share`
 - nixpkgs: `/nix/store/89bzgr2smf8jdqlbay9apn6h81qrp238-source`


### Are you using nix-darwin?

Yes, I am using nix-darwin.

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+bug%22+label%3A%226.topic%3A+darwin%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/llm/default.nix
+++ b/pkgs/development/python-modules/llm/default.nix
@@ -4,6 +4,7 @@
   callPackage,
   buildPythonPackage,
   fetchFromGitHub,
+  fetchpatch,
   pytestCheckHook,
   pythonOlder,
   replaceVars,
@@ -24,6 +25,7 @@
   pytest-asyncio,
   pytest-httpx,
   pytest-recording,
+  sqlite,
   sqlite-utils,
   syrupy,
   llm-echo,
@@ -179,7 +181,17 @@ let
       hash = "sha256-PMQGyBwP6UCIz7p94atWgepbw9IwW6ym60sfP/PBrCA=";
     };
 
-    patches = [ ./001-disable-install-uninstall-commands.patch ];
+    patches = [
+      ./001-disable-install-uninstall-commands.patch
+    ]
+    # See https://github.com/NixOS/nixpkgs/issues/476258 and https://github.com/simonw/llm/pull/1334
+    # TODO: Remove when sqlite 3.52.x is released.
+    ++ lib.optionals (sqlite.version == "3.51.1") [
+      (fetchpatch {
+        url = "https://github.com/simonw/llm/commit/6e24b883c3e3c4ddd2ec9006714d0a9ec17b59da.patch";
+        hash = "sha256-4AKQdZCr6qxuWnjWoSW6I44hPL5e7tnvREx2Ns0WwNc=";
+      })
+    ];
 
     postPatch = ''
       substituteInPlace llm/cli.py \

